### PR TITLE
Test Case for BaseHTTP

### PR DESCRIPTION
**Pull request by jchen236**
Hello Trevor.
Test cases ( a few) !!!!!!!!!!!11!!!

### CODE REVIEW

File: cs-minibot-platform-src/src/main/java/simulator/baseinterface/SimulatorVisionSystem.java [hunk 1]
```diff
@@ -86,6 +84,10 @@ public void stepSimulation() {
             }
     }
 
+    public HashSet<PhysicalObject> getAllPhysicalObjects() {
+        return po_set;
```
Comment (TrevorEdwards) [inline cs-minibot-platform-src/src/main/java/simulator/baseinterface/SimulatorVisionSystem.java]: camelCase

File: cs-minibot-platform-src/src/test/java/BaseHTTPInterfaceTest.java [hunk 2]
```diff
@@ -0,0 +1,154 @@
+/**
+ * Created by jimmychen on 4/20/17.
```
Comment (TrevorEdwards) [inline cs-minibot-platform-src/src/test/java/BaseHTTPInterfaceTest.java]: no author comments

File: cs-minibot-platform-src/src/test/java/BaseHTTPInterfaceTest.java [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+import static spark.Spark.port;
+import static spark.Spark.staticFiles;
+
+import org.junit.runners.MethodSorters;
+import simulator.baseinterface.SimulatorVisionSystem;
+import simulator.physics.PhysicalObject;
+import simulator.simbot.SimBot;
+import simulator.simbot.SimBotConnection;
+import spark.route.RouteOverview;
+
```
Comment (TrevorEdwards) [inline cs-minibot-platform-src/src/test/java/BaseHTTPInterfaceTest.java]: make sure all these are used

File: cs-minibot-platform-src/src/test/java/BaseHTTPInterfaceTest.java [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+import java.io.*;
```
Comment (TrevorEdwards) [inline cs-minibot-platform-src/src/test/java/BaseHTTPInterfaceTest.java]: avoid * imports when possible

File: cs-minibot-platform-src/src/test/java/BaseHTTPInterfaceTest.java [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+            ProcessBuilder pb = new ProcessBuilder("python", "test_tcpProcess.py");
+            pb.start();
+        }
+        catch(Exception e){
+            System.out.println(e);
+        }
+
+        System.out.println("opening TCP");
+        Thread.sleep(5000);
+    }
```
Comment (TrevorEdwards) [inline cs-minibot-platform-src/src/test/java/BaseHTTPInterfaceTest.java]: new line after this

File: cs-minibot-platform-src/src/test/java/BaseHTTPInterfaceTest.java [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+    @Test
+    public void testAddBot() throws Exception {
+
```
Comment (TrevorEdwards) [inline cs-minibot-platform-src/src/test/java/BaseHTTPInterfaceTest.java]: max 1 newline (but none here)

File: cs-minibot-platform-src/src/test/java/BaseHTTPInterfaceTest.java [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+        Bot testMiniBot;
+        Bot testModBot;
+        Bot testSimBot;
+        String ip = "127.0.0.1";
+        int port = 10003;
+
+
+        port(8080);
+        System.out.println("fsfu");
```
Comment (TrevorEdwards) [inline cs-minibot-platform-src/src/test/java/BaseHTTPInterfaceTest.java]: delete

File: cs-minibot-platform-src/src/test/java/BaseHTTPInterfaceTest.java [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+        staticFiles.location("/public");
+        RouteOverview.enableRouteOverview("/");
+        SimulatorVisionSystem simvs;
+        simvs = SimulatorVisionSystem.getInstance();
+        assertNotNull(simvs);
+        assertNotEquals(BaseStation.getInstance().getVisionManager().addVisionSystem(simvs), 0);
+        System.out.println("fldf");
```
Comment (TrevorEdwards) [inline cs-minibot-platform-src/src/test/java/BaseHTTPInterfaceTest.java]: delete

File: cs-minibot-platform-src/src/test/java/BaseHTTPInterfaceTest.java [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+
+//        IceConnection ice = new IceConnection(ip, port);
+//        testModBot = new ModBot(ice, "TestCaseModBot");
```
Comment (TrevorEdwards) [inline cs-minibot-platform-src/src/test/java/BaseHTTPInterfaceTest.java]: delete

File: cs-minibot-platform-src/src/test/java/BaseHTTPInterfaceTest.java [hunk 10]
```diff
@@ -0,0 +1,154 @@
...
+        System.out.println("hi");
```
Comment (TrevorEdwards) [inline cs-minibot-platform-src/src/test/java/BaseHTTPInterfaceTest.java]: delete

File: cs-minibot-platform-src/src/test/java/BaseHTTPInterfaceTest.java [hunk 11]
```diff
@@ -0,0 +1,154 @@
...
+        PhysicalObject po = new PhysicalObject("TESTBOT", 50, simvs.getWorld(), 0.4f, 0.0f, 1f, 1f, true);
+        SimBot smbot;
+        smbot = new SimBot(sbc, "TestCaseSimBot", po);
+        testSimBot = smbot;
+
+        ArrayList<PhysicalObject> pObjs = new ArrayList<>();
+        pObjs.add(po);
+        simvs.processPhysicalObjects(pObjs);
+
+        //assertEquals(BaseStation.getInstance().getBotManager().addBot(testModBot), "TestCaseModBot");
```
Comment (TrevorEdwards) [inline cs-minibot-platform-src/src/test/java/BaseHTTPInterfaceTest.java]: delete

File: cs-minibot-platform-src/src/test/java/BaseHTTPInterfaceTest.java [hunk 12]
```diff
@@ -0,0 +1,154 @@
...
+        assertEquals(BaseStation.getInstance().getBotManager().addBot(testMiniBot), "TestCaseMiniBot");
+        assertEquals(BaseStation.getInstance().getBotManager().addBot(testSimBot), "TestCaseSimBot");
+
+        Optional empty = Optional.empty();
+        //assertNotEquals(BaseStation.getInstance().getBotManager().getBotByName("TestCaseModBot"), empty);
```
Comment (TrevorEdwards) [inline cs-minibot-platform-src/src/test/java/BaseHTTPInterfaceTest.java]: delete

File: cs-minibot-platform-src/src/test/java/BaseHTTPInterfaceTest.java [hunk 13]
```diff
@@ -0,0 +1,154 @@
...
+        System.out.println("hi");
+        TCPConnection c = new TCPConnection(ip, port);
+        testMiniBot = new MiniBot(c, "TestCaseMiniBot");
+        assertEquals(c.connectionActive(), true);
+
+        SimBotConnection sbc = new SimBotConnection();
+        assertEquals(sbc.connectionActive(), true);
+        PhysicalObject po = new PhysicalObject("TESTBOT", 50, simvs.getWorld(), 0.4f, 0.0f, 1f, 1f, true);
+        SimBot smbot;
+        smbot = new SimBot(sbc, "TestCaseSimBot", po);
```
Comment (TrevorEdwards) [inline cs-minibot-platform-src/src/test/java/BaseHTTPInterfaceTest.java]: a lot is being done in this test. i'll let it slide but in general try to keep unit tests as discrete as possible

File: cs-minibot-platform-src/src/test/java/BaseHTTPInterfaceTest.java [hunk 14]
```diff
@@ -0,0 +1,154 @@
...
+        assertEquals(BaseStation.getInstance().getBotManager().addBot(testMiniBot), "TestCaseMiniBot");
+        assertEquals(BaseStation.getInstance().getBotManager().addBot(testSimBot), "TestCaseSimBot");
+
+        Optional empty = Optional.empty();
+        //assertNotEquals(BaseStation.getInstance().getBotManager().getBotByName("TestCaseModBot"), empty);
+        assertTrue(BaseStation.getInstance().getBotManager().getBotByName("TestCaseMiniBot").isPresent());
+        assertTrue(BaseStation.getInstance().getBotManager().getBotByName("TestCaseSimBot").isPresent());
+
+        assertEquals(simvs.getAllPhysicalObjects().size(), 1);
+
```
Comment (TrevorEdwards) [inline cs-minibot-platform-src/src/test/java/BaseHTTPInterfaceTest.java]: delete

File: cs-minibot-platform-src/src/test/java/BaseHTTPInterfaceTest.java [hunk 15]
```diff
@@ -0,0 +1,154 @@
...
+    }
```
Comment (TrevorEdwards) [inline cs-minibot-platform-src/src/test/java/BaseHTTPInterfaceTest.java]: newline after this line

**Comment by alan20854** [comment]
lol

File: cs-minibot-platform-src/src/test/java/BaseHTTPInterfaceTest.java [hunk 16]
```diff
@@ -0,0 +1,154 @@
...
+//            scriptOut.close();
+//
+//            ProcessBuilder pb = new ProcessBuilder("python", "test_close_tcp.py");
+//            pb.start();
+//        }
+//        catch(Exception e){
+//            System.out.println(e);
+//        }
+
+        System.out.println("closing TCP");
```
Comment (TrevorEdwards) [inline cs-minibot-platform-src/src/test/java/BaseHTTPInterfaceTest.java]: just comment this too

File: test_tcpProcess.py [hunk 17]
```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/python
+
+from socket import *
+import multiprocessing, time, signal, os, sys, threading, socket
```
Comment (TrevorEdwards) [inline test_tcpProcess.py]: delete file

File: test_close_tcp.py [hunk 18]
```diff
@@ -0,0 +1,13 @@
+from socket import *
+import multiprocessing, time, signal, os, sys, threading, socket
+
+#From https//github.com/zephod/legopi
+# from lib.legopi.lib import xbox_read
+
+
+from multiprocessing import Process
```
Comment (TrevorEdwards) [inline test_close_tcp.py]: move this file somewher eelse

File: test_tcpProcess.py (hunk 17 above)
Comment (TrevorEdwards) [inline test_tcpProcess.py]: stil needs delet?

Comment (jchen236) [inline test_tcpProcess.py]: DELETed

**Comment by TrevorEdwards** [comment]
merging, but had to disable the simulator cases so i could get this done quickly

**Pull request merged**